pythnet/pythnet_sdk/src/messages.rs

```diff
@@ -175,6 +177,27 @@ impl Arbitrary for TwapMessage {
     }
 }
 
+/// The time weighted average price & conf for a feed over the window [start_time, end_time].
+/// This type is used to persist the calculated TWAP in TwapUpdate accounts on Solana.
+#[repr(C)]
+#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize, BorshSchema)]
+#[cfg_attr(feature = "solana-program", derive(AnchorSerialize, AnchorDeserialize))]
+#[cfg_attr(
+    not(feature = "solana-program"),
+    derive(BorshSerialize, BorshDeserialize)
+)]
+pub struct TwapPrice {
+    pub feed_id: FeedId,
+    pub start_time: i64,
+    pub end_time: i64,
+    pub price: i64,
+    pub conf: u64,
+    pub exponent: i32,
+    /// Ratio out of 1_000_000, where a value of 1_000_000 represents
+    /// all slots were missed and 0 represents no slots were missed.
+    pub down_slots_ratio: u32,
+}
```

Comment:
I have a mixed feeling about this being here because Pythnet SDK mostly represents things that are generated and come out of it and it is mostly a derivation that is only used externally from the Solana receiver (and users will still use the re-exported struct from the solana sdk). If we ever expand to other rust-based networks we might need to re-use it but they might end up having different layouts (like Hermes has its own layout).

Reply:
sure thing -- was following the example of `PriceFeedMessage`, but will move this back to the solana receiver SDK